src/header/common/origin.rs

```diff
@@ -0,0 +1,125 @@
+use header::{Header, Host, HeaderFormat};
+use std::fmt;
+use url::Url;
+use header::parsing::from_one_raw_str;
+
+/// The `Origin` header.
+///
+/// The `Origin` header is a version of the `Referer` header that is used for all HTTP fetches and `POST`s whose CORS flag is set.
+/// This header is often used to inform recipients of the security context of where the request was initiated.
+///
+///
+/// Following the spec, https://fetch.spec.whatwg.org/#origin-header, the value of this header is composed of
+/// a String (scheme), header::Host (host/port)
+///
+/// # Examples
+/// ```
+/// use hyper::header::{Headers, Origin, Host};
+///
+/// let mut headers = Headers::new();
+/// headers.set(
+///     Origin{
+///           scheme: "http".to_owned(),
+///           host: Host{hostname: "hyper.rs".to_owned(), port: None}
+///    }
+/// );
+/// ```
+/// ```
+/// use hyper::header::{Headers, Origin, Host};
+///
+/// let mut headers = Headers::new();
+/// headers.set(
+///     Origin{
+///           scheme: "https".to_owned(),
+///           host: Host{hostname: "wikipedia.org".to_owned(), port: Some(443)}
+///     }
+/// );
+/// ```
+
+#[derive(Clone, Debug)]
+pub struct Origin {
+    /// The scheme, such as http or https
+    pub scheme: String,
+    /// The host, such as Host{hostname: "hyper.rs".to_owned(), port: None}
+    pub host: Host,
```

Comment:
It might not be important yet. The idea I had was to have `scheme()`, `hostname()`, and `port()` accessors. By keeping the fields themselves private, it doesn't matter if we keep using `Host` internally, or some other field scheme.

But all the other headers are currently wide open. I'll consider a accessor pass before 1.0.